### ISSUE TITLE
Transport resource not being found

### ISSUE THREAD

**Issue by willonit** [vmware/vmware-vcenter#219]
When I install the module on a node and `puppet apply` the following code everything works fine.
```
transport{'vcenter':
	username => 'administrator@vsphere.local',
	password => $pass,
	server	 => 'vcenter.domain.local',
	options  => { 'insecure' => true },
}
vc_folder{'/DC1/Folder':
	ensure => present,
	transport => Transport['vcenter'],
}
```

When the code is moved to a master and a `puppet agent -t` is run, the following error is thrown. 

`Error: /Stage[main]/Profile::Vsphere/Vc_folder[/DC/Folder]: Could not evaluate: Invalid transport Transport[vcenter] provided for Vc_folder[/DC/Folder]`

I'm unsure where to begin to troubleshoot this further. Is anyone else using this in a master, agent configuration? Has this error been seen before?

**Comment by ishashchuk** [comment]
@willonit Did you get it figured out by any chance? I'm having the same issues in master/agent configuration and can't figure out what is the problem. Puppet apply works with no issues

**Comment by willonit** [comment]
I do not believe this module is designed to work in a master/agent scenario. I never could get it going. I ended up using the master to clone a repo with the code that I wanted and exec a puppet apply. Not pretty by any means but closest I could get to the desired outcome.
